test/system/src/main/java/io/pravega/test/system/framework/DockerRemoteSequential.java

```diff
@@ -0,0 +1,154 @@
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ContainerCreation;
+import com.spotify.docker.client.messages.ContainerInfo;
+import com.spotify.docker.client.messages.HostConfig;
+import com.spotify.docker.client.messages.PortBinding;
+import io.pravega.common.concurrent.FutureHelpers;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.NotImplementedException;
+import org.junit.Assert;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import static io.pravega.test.system.framework.Utils.getConfig;
+import static org.junit.Assert.assertFalse;
+
+@Slf4j
+public class DockerRemoteSequential implements TestExecutor {
+
+    public final static DockerClient CLIENT = DefaultDockerClient.builder().uri("http://"+ getConfig("masterIP", "Invalid Master IP")+ ":2375").build();
+    private final static String IMAGE = "java:8";
+    public String id;
+    private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(3);
+
+    public CompletableFuture<Void> startTestExecution(Method testMethod) {
+
+        log.debug("Starting test execution for method: {}", testMethod);
+
+        String className = testMethod.getDeclaringClass().getName();
+        String methodName = testMethod.getName();
+        String containerName = methodName + ".testJob";
+
+        return CompletableFuture.runAsync(() -> {
+            startTest(containerName, className, methodName);
+        }).thenCompose(v2 -> waitForJobCompletion(containerName))
+                .whenComplete((v, ex) -> {
+            stopContainer(); //stop container once execution is complete.
+                    if (ex != null) {
+                        log.error("Error while executing the test. ClassName: {}, MethodName: {}", className,
+                                methodName);
+                    }
+                }
+                );
+    }
+
+
+    @Override
+    public CompletableFuture<Void> stopTestExecution(String testID) {
+        throw new NotImplementedException("Stop Execution is not used for Remote sequential execution");
+    }
+
+    private CompletableFuture<Void> waitForJobCompletion(final String containerName) {
+        return FutureHelpers.loop(() -> isTestRunning(containerName),
+                () -> FutureHelpers.delayedFuture(Duration.ofSeconds(3), executorService),
+                executorService);
+    }
+
+    private boolean isTestRunning(String containerName) {
+        boolean value = false;
+        try {
+            ContainerInfo info = CLIENT.inspectContainer(id);
+                if (info.name().equals(containerName)) {
```

Comment:
Weird indentation in this file. Run it through an autoformatter.

Reply:
done.